Loop/Managers/DeviceDataManager.swift

```diff
@@ -871,6 +851,36 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
 
     let doseStore: DoseStore
 
+    // MARK: DoseStoreDelegate
+
+    func doseStore(doseStore: DoseStore, hasEventsNeedingUpload pumpEvents: [(objectID: NSManagedObjectID, date: NSDate, dose: DoseEntry?, raw: NSData?)], withCompletion completionHandler: (uploadedObjects: [NSManagedObjectID]) -> Void) {
+
+        guard let uploader = remoteDataManager.nightscoutUploader, device = rileyLinkManager.firstConnectedDevice, pumpModel = pumpState?.pumpModel else {
```

Comment:
The only use I've ever had for it was when using multiple RLs; and seeing when one took over.  PumpID is not uploaded, but I could certainly see that being useful if you were switching pumps.   Switching pumps is a pretty complicated scenario; would be difficult to offer full support for it, I think, but things like tracking which pump an event came from would certainly help.